rest-service/manager_rest/rest/resources_v3_1/blueprints.py

```diff
@@ -327,37 +320,34 @@ def patch(self, blueprint_id, **kwargs):
                     cleanup_blueprint_archive_from_file_server(
                         blueprint_id=blueprint_id,
                         tenant=blueprint.tenant.name)
-        else:  # Updating the blueprint not as part of the upload process
-            if provided_labels is not None:
-                if blueprint.state != BlueprintUploadState.UPLOADED:
-                    raise ConflictError(
-                        'Blueprint labels can only be updated if the blueprint'
-                        ' was uploaded successfully')
 
-                rm = get_resource_manager()
-                labels_list = rest_utils.get_labels_list(provided_labels)
-                rm.update_resource_labels(models.BlueprintLabel,
-                                          blueprint,
-                                          labels_list,
-                                          creator=creator,
-                                          created_at=created_at)
+        labels_list = None
+        if state == BlueprintUploadState.UPLOADED:
+            if request_dict.get('labels'):
+                labels_list = [Label(**label)
+                               for label in request_dict.get('labels')]
```

Comment:
```suggestion
                               for label in request_dict['labels']]
```
the .get is redundant here